oada-core/http-handler/oada-srvc-http-handler/resources.js

```diff
@@ -1,239 +1,282 @@
+const uuid = require('uuid')
+const axios = require('axios')
+const express = require('express')
+const bodyParser = require('body-parser')
+const pointer = require('json-pointer')
+const _ = require('lodash')
+const typeis = require('type-is')
+const { pipeline } = require('stream')
+const pipelineAsync = require('bluebird').promisify(pipeline)
+const cacache = require('cacache')
+
+const info = require('debug')('http-handler:info')
+const warn = require('debug')('http-handler:warn')
+const error = require('debug')('http-handler:error')
+const trace = require('debug')('http-handler:trace')
+
+const resources = require('../../libs/oada-lib-arangodb').resources
+const changes = require('../../libs/oada-lib-arangodb').changes
+const putBodies = require('../../libs/oada-lib-arangodb').putBodies
+const OADAError = require('oada-error').OADAError
+
+const config = require('./config')
+const CACHE_PATH = config.get('storage:binary:cacache')
+
+var requester = require('./requester')
+
+var router = express.Router()
+var expressWs = require('express-ws')(router)
 
 // Turn POSTs into PUTs at random id
-router.post('/*?', function postResource(req, res, next) {
-    // TODO: Is this a good way to generate new id?
-    if (req.url === '' || req.url.endsWith('/')) {
-        req.url += uuid();
-    } else {
-        req.url += '/' + uuid();
-    }
-    req.method = 'PUT';
+router.post('/*?', function postResource (req, res, next) {
+  // TODO: Is this a good way to generate new id?
+  if (req.url === '' || req.url.endsWith('/')) {
+    req.url += uuid()
+  } else {
+    req.url += '/' + uuid()
+  }
+  req.method = 'PUT'
 
-    next();
-});
+  next()
+})
 
-router.use(function graphHandler(req, res, next) {
-    Promise.resolve(
-        resources.lookupFromUrl("/resources" + req.url, req.user.user_id),
+router.use(function graphHandler (req, res, next) {
+  Promise.resolve(
+    resources.lookupFromUrl('/resources' + req.url, req.user.user_id)
+  )
+    .then(function handleGraphRes (resp) {
+      info('GRAPH LOOKUP RESULT', resp)
+      if (resp['resource_id']) {
+        // Rewire URL to resource found by graph
+        let url = `${resp['resource_id']}${resp['path_leftover']}`
+        // log
+        console.log(`Graph lookup: ${req.url} => ${url}`)
+        // Remove "/resources" from id
+        req.url = url.replace(/^\/?resources\//, '/')
+      }
+      res.set('Content-Location', req.baseUrl + req.url)
+      // TODO: Just use express parameters rather than graph thing?
+      req.oadaGraph = resp
+      req.resourceExists = _.clone(resp.resourceExists)
+    })
+    .asCallback(next)
+})
+
+router.delete('/*', function checkScope (req, res, next) {
+  requester
+    .send(
+      {
+        connection_id: req.id,
+        domain: req.get('host'),
+        oadaGraph: req.oadaGraph,
+        user_id: req.user['user_id'],
+        scope: req.user.scope,
+        contentType: req.get('content-type'),
+        requestType: 'delete'
```

Comment:
This diff looks big because my nvim autofixed the format. The line is the only actual change to the file.

The logical diff for the line is this:
```diff
- requestType: 'put'
+ requestType: 'delete'
```

Reply:
Formating looks better :)